kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -38,6 +38,15 @@ import scala.collection.JavaConverters._
 
 class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
 
+  test("CRC metadata & protocol fetching") {
+    val path = getTestResourceFilePath("stream_table_optimize")
```

Comment:
cc @allisonport-db -- does kernel support golden tables? i.e. defining the actual scala/spark code to create the resource table?

For example, how was this table created? What was the code? I can't see that from looking at the resources folder.

Reply:
I believe it is the static files in the folder `kernel/kernel-defaults/src/test/resources/stream_table_optimize` I think I could make a follow up PR for switching to golden tables considering many added test cases depends on this added table.

Reply:
Now that Delta 3.3 writes crc files we can either use golden tables, or write the test table in the tests (if it's reused a lot maybe using a golden table is preferable)

Reply:
Update the tests using spark to generate tables